tests/components/shelly/test_sensor.py

```diff
@@ -237,7 +247,9 @@ async def test_block_not_matched_restored_sleeping_sensor(
     assert hass.states.get(entity_id).state == "20.4"
 
     # Make device online
-    monkeypatch.setattr(mock_block_device.blocks[SENSOR_BLOCK_ID], "type", "other_type")
+    monkeypatch.setattr(
+        mock_block_device.blocks[SENSOR_BLOCK_ID], "description", "other_desc"
```

Comment:
This was the reason I originally assumed unique ids of the device can't be used for correlating back to the entity descriptions.

We store the unique id using `block.description` - https://github.com/home-assistant-libs/aioshelly/blob/b724677aec88ba840956c58f900e3ac00fb81fbc/aioshelly/block_device/device.py#L487

When creating entities we use `block.type` - https://github.com/home-assistant-libs/aioshelly/blob/b724677aec88ba840956c58f900e3ac00fb81fbc/aioshelly/block_device/device.py#L436 (to allow multiple instances of a sensor) to fetch entity descriptions

However they are build in `aioshelly` from the same `self.blk["D"]`